### PR TITLE
Generic serializer +  spray support

### LINKED ISSUE #24: Remove circe dependency from `call`
Figure out how to do  generic encoder/decoder and add a binding to circe as a separate project.

https://github.com/erdeszt/graphient/blob/master/graphient/src/main/scala/graphient/GraphientClient.scala#L32

### PR DESCRIPTION
**Pull request by erdeszt**
Generic `Encoder` & `Decoder` interfaces and support for both `circe` and `spray`. Fixes #24 and #29

### CODE REVIEW

**Comment by erdeszt** [comment]
~TODO: Check if `README.md` and is up to date, add more detailed instructions~ Readme updated.

**Comment by erdeszt** [comment]
Fix organization name for spray & circe projects

File: build.sbt [hunk 1]
```diff
@@ -1,58 +1,77 @@
...
+    scalaVersion := "2.12.6",
+    version := "1.0.0",
+    scalacOptions ++= ScalacOptions,
+    resolvers += Resolver.sonatypeRepo("releases"),
+    licenses += ("MIT", url("http://opensource.org/licenses/MIT")),
+    libraryDependencies ++= circe,
+    libraryDependencies ++= scalaTest.map(_ % Test),
+  ).dependsOn(graphientCore % "compile->compile;test->test")
+
+lazy val graphientSpray = (project in file("graphient-spray"))
```
Comment (pellekrogholt) [inline build.sbt:66]: I needs an organization `organization := "io.github.erdeszt",` to get the publish work - I tried out locally with `sbt graphientSpray/publishLocal`

and without organization i published it to the wrong spot:

`/home/pelle/.ivy2/local/graphient-spray/graphient-spray_2.12/1.0.0-SNAPSHOT/ivys/ivy.xml`

consumers look for it here

`/home/pelle/.ivy2/local/io.github.erdeszt/graphient-spray_2.12/1.0.0-SNAPSHOT/ivys/ivy.xml`

**Comment by erdeszt** [comment]
Double check documentation. `graphient.model._`

**Comment by erdeszt** [comment]
Fix documentation for serializer namespaces

**Review by pellekrogholt — APPROVED** [review]
LGTM. Great work! Tried it out with a fresh domain specific project :ship:

**Pull request merged**